asynciterator.ts

```diff
@@ -1251,6 +1259,104 @@ function destinationFillBuffer<S>(this: InternalSource<S>) {
     (this._destination as any)._fillBuffer();
 }
 
+interface ComposedFunction {
+  fn: Function,
+  close?: boolean,
+  next?: ComposedFunction
+}
+
+export class MappingIterator<T, D = T> extends AsyncIterator<D> {
+  private _fn?: Function;
+  private _destroySource: boolean;
+  private onSourceError = (err: Error) => this.emit('error', err);
+  private onSourceReadable = () => this.emit('readable');
+  private onSourceEnd = () => this.close();
```

Comment:
We can avoid creating and storing these as follows: https://github.com/RubenVerborgh/AsyncIterator/blob/829087fb5f9add230aa09606c6a964ef1189fbd4/asynciterator.ts#L1124-L1135